services/ui-src/src/App.tsx

```diff
@@ -171,6 +151,68 @@ export function App() {
     setUserInfo();
   }, [setUserInfo]);
 
+  useEffect(() => {
+    if (mmdlNotification !== undefined) {  // Ensure the flag has been resolved
+      setNotificationState(mmdlNotification);
+    }
+  }, [mmdlNotification]);
```

Comment:
Having this part of the code be in its own useEffect seems like extra work/slows down the app

How about instead, move this line of code 
`const {mmdlNotification} = useFlags();` into the useEffect on line 160, and get rid of the dependencies in the array at the end of the useEffect. that way this will always run on the start of the app